### ISSUE TITLE
Pop-up style "When mouse moves to top of screen" has unobvious hardcoded behaviours

### ISSUE THREAD

**Issue by morganwillcock** [adventuregamestudio/ags#530]
Mainly discussed here, as to why closing automatically when clicked is bad and confusing: [http://www.adventuregamestudio.co.uk/forums/index.php?topic=56461.0](http://www.adventuregamestudio.co.uk/forums/index.php?topic=56461.0l)

...but as another example of why how this works is problematic, it also switches the cursor mode by itself. The BASS template only has a single cursor defined, so by switching the GUI to use the built-in pop-up mode it makes the cursor disappear.

![lost_cursor](https://user-images.githubusercontent.com/14958747/48033030-9bf62300-e151-11e8-993d-2b56a6803211.gif)

Suggestion is (AGS4 only?):
- Add auto-close option to GUIs, if you want them to close after a single (handled?) click
- Don't override cursor modes based on a display option - how it is opened is potentially unrelated to what its intended function is

**Comment by ivan-mogilko** [comment]
I guess that's a matter of what they call "software philosophy", but since we do not have any convention established for AGS, then my personal subjective stance on this is: if something may be easily done by script then it should not be added as a builtin behavior.

For that reason I do not like auto-close option (besides it will have very narrow use).

In fact, IMHO the main thing about this gui style is that it is obsolete on its own as a builtin mechanic since you may script same behavior, and you will have to do that anyway if you want to, for example, change the screen side it slides from. If not for backwards compatibility, I might even suggest to deprecate and remove it from AGS 4, and instead provide some script module for most useful GUI behaviors.

**Comment by morganwillcock** [comment]
>I'd rather vote for expanding script handling of clicks on gui, - that would cover far more use-cases

Yes. Removing it entirely would be fine, following a the same principle, someone can change a cursor mode and close the GUI with a script, and keeping it in is confusing for anyone not realising it is for a very specific use case.

For making it work differently right now, I'm not sure. It seems so focused on the original interface design that anyone using genuinely using it is probably looking for this exact behavior.
